artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/broker/AMQPMessage.java

```diff
@@ -624,18 +626,18 @@ public AMQPMessage setAddress(SimpleString address) {
 
    private void setProtonAddress(String address) {
       MessageImpl protonMessage = getProtonMessage();
-      if (protonMessage.getProperties() == null) {
-         protonMessage.setProperties(new Properties());
+      if (protonMessage.getDeliveryAnnotations() == null) {
+         protonMessage.setDeliveryAnnotations(new DeliveryAnnotations(new HashMap<>()));
       }
-      protonMessage.getProperties().setTo(address.toString());
+      protonMessage.getDeliveryAnnotations().getValue().put(AMQP_HDR_ORIGINAL_ADDRESS, address);
    }
 
    @Override
    public SimpleString getAddressSimpleString() {
       if (address == null) {
-         Properties properties = getProtonMessage().getProperties();
-         if (properties != null) {
-            internalSetAddress(properties.getTo());
+         DeliveryAnnotations deliveryAnnotations = getProtonMessage().getDeliveryAnnotations();
+         if (deliveryAnnotations != null && deliveryAnnotations.getValue() != null && deliveryAnnotations.getValue().containsKey(AMQP_HDR_ORIGINAL_ADDRESS)) {
+            internalSetAddress((String) deliveryAnnotations.getValue().get(AMQP_HDR_ORIGINAL_ADDRESS));
```

Comment:
This bit changing to only look at the DeliveryAnnotations mane the address field in AMQPMessage wont be getting set based on the Properties to field as it would before, is that intended?

Reply:
yes that';s intentional.

Reply:
@gemmellr @clebertsuconic I am going to look at using extra properties for this.  As a general comment we can't rely on the To field of the AMQP message.  This has been the reason for a lot of clustering issues recently.  We should be able to persist some meta-data that has the original address.  Either from the link or the message to field.

Reply:
I will go for extraProperties